### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'args'

### ISSUE THREAD

**Issue by frapik99** [angr/angr#432]
Hi,
on Windows it not possible to pass args to ```full_init_state```. How can I pass parameter to the state?

Thanks

**Comment by rhelmot** [comment]
This can't possibly have anything to do with Windows. Can you post the exact code you're using?

**Comment by frapik99** [comment]
@rhelmot , I have use the following line:

```
state = b.factory.full_init_state(args=['file.exe', claripy.BVS('arg1', 20*8)])
```

Thanks

**Comment by rhelmot** [comment]
Oh!! You must be trying to analyze a Windows *program*. We don't support passing arguments to windows programs. If you want this to work you will need to implements a SimWindows as a subclass of SimOS, teach it how to serialize args into the state, and register it in simos.py.

**Comment by frapik99** [comment]
Thanks @rhelmot , is there an easier way for doing that? Maybe a callable state?

Thanks

**Comment by schieb** [comment]
FYI: I'm working on some related stuff so if you don't feel like implementing that then just sit tight for a week or two for a PR that will probably solve your problem. I'll try to remember to reference this when I open the PR, so feel free to leave open.

In the meantime you could set up the args manually and set entry address to `main`. Pay attention to the calling convention, especially if you are on x64.

**Comment by frapik99** [comment]
Thanks @schieb , I'll try as you said and also wait for the PR.

Thanks

**Comment by schieb** [comment]
This was fixed in #618
